llvm/lib/Target/SPIRV/SPIRVEmitIntrinsics.cpp

```diff
@@ -845,6 +846,17 @@ Type *SPIRVEmitIntrinsics::deduceElementTypeHelper(
       } else {
         llvm_unreachable("Unknown handle type for spv_resource_getpointer.");
       }
+    } else if (II &&
+               II->getIntrinsicID() == Intrinsic::spv_pushconstant_getpointer) {
+      auto *GV = cast<GlobalVariable>(II->getOperand(0));
+      auto *HandleType = cast<TargetExtType>(GV->getValueType());
+      if (HandleType->getTargetExtName() == "spirv.PushConstant") {
```

Comment:
should this be an assert or error of some type? What else could the handle type be?